### ISSUE TITLE
Suspend project when subscription cancelled in Stripe

### ISSUE THREAD

**Issue by ZJvandeWeg** [flowforge/flowforge#1162]
### Current Behavior

When payments fail there's currently no logic to suspends a teams projects that consume resources.

### Expected Behavior

After two payments for the same invoice fail:
- Suspend all projects of that team
- Suspend all devices and their updates
- Downgrade the team type (if possible)

### Environment

- FlowForge version:
- Node.js version:
- npm version:
- Platform/OS:
- Browser:

**Comment by Pezmc** [comment]
@ZJvandeWeg To check the business use case here. This is to reduce the amount of time that teams projects are left running when the subscription is in a `past_due` state?

I ask as Stripe has an explicit subscription cancelled state that you'd usually tie to locking the customers subscription, in FlowForges case suspending projects, devices and downgrading team type, that's configured as part of smart retry.

<img width="1601" alt="Screenshot 2022-12-07 at 15 11 29" src="https://user-images.githubusercontent.com/507155/206190124-9c7a6c8b-6d57-430b-8ee9-7e21b9cbf127.png">

Is this feature designed to be on top of handling the subscription cancelled event i.e:

 - Charge fails
 - Subscription goes into past due state
 - Charge fails again
 - FlowForge shuts down customer projects
 - Charge continues to fail following Stripe auto retry logic
 - Eventually subscription is cancelled 
 - FlowForge handles subscription cancel

Or a replacement for it?

**Comment by ZJvandeWeg** [comment]
For now we leave the Stripe settings as is and we implement new logic to handle the 'cancelled' event from Stipe. There's manual work to follow up on; we need to suspend all cancelled subscriptions.

**Comment by knolleary** [comment]
Unclear what the status of this issue is in the 1.2 release as the issue is still open with no updates, but is in the Verify state.

There are (at time of this comment) 4 open issues referencing this one. Most read as if they are future work, but #1462 mentions updates needed in production which haven't been flagged for anyone to take care of in the release.

**Comment by knolleary** [comment]
Discussing with @hardillb, I believe the core requirement has been satisfied for 1.2, but with some residual follow-up work identified. I'm going to bump this issue to 1.3 so it doesn't drop off the plan before we can confirm this issue can be closed.

**Comment by Pezmc** [comment]
@knolleary Yes, the core work has been completed before I left and I’ve split any (optional) follow up into separate issued tagged above: #1457, #1463 , #1457

**Comment by Pezmc** [comment]
None of the remaining issues are _required_ to consider this work complete, so I'm marking this issue done. Any follow up work to the billing UI or UX around cancellations can be tracked separately.
